modules/core/src/ocl.cpp

```diff
@@ -5166,6 +5241,144 @@ MatAllocator* getOpenCLAllocator()
     return allocator;
 }
 
+
+//////////////////////////////////////////////////////////////////////////////
+// static helper funcs (dups from umatrix.cpp)
+
+static inline void setSize(UMat& m, int _dims, const int* _sz,
```

Comment:
Code duplication is bad idea. Try to move all pieces (similar code for Mat) into header file